src/server_manager/web_app/ui_components/outline-server-view.js

```diff
@@ -654,61 +661,67 @@ Polymer({
     } else if (event.key === 'Enter') {
       input.blur();
     }
-  },
+  }
 
-  _handleNameInputBlur: function(event) {
+  _handleNameInputBlur(event) {
     const input = event.target;
     const accessKey = event.model.item;
     const displayName = input.value;
     if (displayName === accessKey.name) {
       return;
     }
     input.disabled = true;
-    this.fire('RenameAccessKeyRequested', {
-      accessKeyId: accessKey.id,
-      newName: displayName,
-      entry: {
-        commitName: () => {
-          accessKey.name = displayName;
-          input.disabled = false;
-        },
-        revertName: () => {
-          input.value = accessKey.name;
-          input.disabled = false;
+    this.dispatchEvent(new CustomEvent('RenameAccessKeyRequested', {
```

Comment:
May be worth decomposing event dispatching into a helper function to avoid duplicating some options (i.e. `bubbles: true`, `composed: true`).

Reply:
Agreed. I created `makePublicEvent()`, which bubbles up beyond the shadow root.